jplephem/spk.py

```diff
@@ -48,7 +48,16 @@ def close(self):
         self.daf.file.close()
         for segment in self.segments:
             if hasattr(segment, '_data'):
-                del segment._data  # TODO: explicitly close each memory map
+                del segment._data
+        try:
```

Comment:
Instead of `try:`, see if this can more efficiently be `if self.daf._array is not None:` and see what happens.